std/functional.d

```diff
@@ -1854,21 +1882,22 @@ if (isCallable!(F))
     }
 }
 
-///
-@system unittest
+@safe unittest
 {
     static int inc(ref uint num) {
         num++;
         return 8675309;
     }
 
-    uint myNum = 0;
-    auto incMyNumDel = toDelegate(&inc);
-    auto returnVal = incMyNumDel(myNum);
-    assert(myNum == 1);
+    uint myNum = 0x1337;
+    struct S1 { int opCall() { inc(myNum); return myNum; } }
+    static assert(!is(typeof(&s1.opCall) == delegate));
+    S1 s1;
```

Comment:
What’s that doing?
In `static assert(!is(typeof(&s1.opCall) == delegate));`, the local variable `s1` isn’t in scope.
If you swap the lines, the `static assert` passes.

Reply:
good catch

Reply:
Apparently, it’s a copy&paste of this code: <https://github.com/dlang/phobos/blob/8c6fca9ccdb30b69b627808c698526a76f7c4248/std/functional.d#L1891-L1901>
<https://github.com/dlang/phobos/blame/8c6fca9ccdb30b69b627808c698526a76f7c4248/std/functional.d#L1891-L1901>